docs/SmartStore.md

```diff
@@ -0,0 +1,233 @@
+  
+
+## Creating a Standalone instance utilizing SmartStore 
+1. Confirm your S3-based storage volume path and URL.
+2. Confirm the name of the Splunk indexes being used with the SmartStore volume. 
+3. Pass the volume and index configuration into the environment through the Standalone Customer Resource specification.
+
+
+Example:
+
+```yaml
+apiVersion: enterprise.splunk.com/v1beta1
+kind: Standalone
+metadata:
+  name: <name>
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  replicas: 1
+  smartstore:
+    volumes:
+    - name: <remote_volume_name>
+         path: <remote_volume_path>
+         endpoint: https://s3-<region>.amazonaws.com
+         secretRef: <secret_store_obj>
+    indexes:
+      - name: <index_name_1>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_2>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_3>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+```
+
+The SmartStore parameters will be placed into the required .conf files in an app. The app is named as `splunk-operator`. In the case of a standalone deployment, the app is located at `/opt/splunk/etc/apps/`
+ 
+ 
+ 
+## Creating an IndexCluster instance utilizing SmartStore
+1. Create an IndexCluster. See [Examples](Examples.md).
+2. Confirm the name of the Splunk indexes being used with the SmartStore volume. 
+   Note: SmartStore configuration is defined only on the Cluster master.
+3. Pass the volume and index configuration into the environment through the ClusterMaster Custom Resource spec.
+
+```yaml
+apiVersion: enterprise.splunk.com/v1beta1
+kind: ClusterMaster
+metadata:
+  name: <name>
+  finalizers:
+  - enterprise.splunk.com/delete-pvc
+spec:
+  smartstore:
+    volumes:
+    - name: <remote_volume_name>
+         path: <remote_volume_path>
+         endpoint: https://s3-<region>.amazonaws.com
+         secretRef: <secret_store_obj>
+    indexes:
+      - name: <index_name_1>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_2>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+      - name: <index_name_3>
+        remotePath: $_index_name
+        volumeName: <remote_volume_name>
+```
+
+The SmartStore parameters will be placed into the required .conf files in an app. The app is named as `splunk-operator`. In case of a Indexer cluster deployment, the app is located on Cluster master at `/opt/splunk/etc/master-apps/`. 
+Once the SmartStore configuration is populated to Cluster Master's `splunk-operator` app, Operator issues a bundle push command to CM, so that the SmartStore configuration is distributed to all the peers in that indexer cluster
```

Comment:
Are we using Cluster Master and CM? Should we use ClusterMaster(CM) in the first occurrence and subsequently use CM?

Reply:
You can always use the "ClusterMaster(CM) in the first occurrence and subsequently use CM" option. But spelling it out is never a bad option for this implied audience.

Reply:
Thanks for clarifying. I am leaving it as "Cluster Master" and "Custom Resource" .